lib/backend/artifactorybackend/blobclient.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2016-2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package artifactorybackend
+
+import (
+	"errors"
+	"fmt"
+	"github.com/uber/kraken/core"
+	"github.com/uber/kraken/lib/backend"
+	"github.com/uber/kraken/lib/backend/artifactorybackend/security"
+	"github.com/uber/kraken/lib/backend/backenderrors"
+	"github.com/uber/kraken/utils/httputil"
+	"github.com/uber/kraken/utils/log"
+	"gopkg.in/yaml.v2"
+	"io"
+	"net/http"
+	"strconv"
+)
+
+const _registryblob = "artifactory_blob"
+
+func init() {
+	backend.Register(_registryblob, &blobClientFactory{})
+}
+
+type blobClientFactory struct{}
+
+func (f *blobClientFactory) Create(
+	confRaw interface{}, authConfRaw interface{}) (backend.Client, error) {
+
+	confBytes, err := yaml.Marshal(confRaw)
+	if err != nil {
+		return nil, errors.New("marshal hdfs config")
+	}
+	var config Config
+	if err := yaml.Unmarshal(confBytes, &config); err != nil {
+		return nil, errors.New("unmarshal hdfs config")
+	}
+	return NewBlobClient(config)
+}
+
+const _layerquery = "http://%s/v2/%s/blobs/sha256:%s"
+//const _layerquery = "http://%s/artifactory/docker/%s/sha256__%s"
```

Comment:
remove?

Reply:
Artifactory supports both APIs. Thus I leave it here for future reference. I will add comments here.

Reply:
instead of commenting out, could you just put it in the comment with more details?